File: 品味Java的21種設計模式/dp_cpp/adapter/example5/ClassAdapter.cpp
#include "ClassAdapter.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace adapter
			{
				namespace example5
				{

					ClassAdapter::ClassAdapter(std::string logFilePathName) : LogFileOperate(logFilePathName)
					{
					}

					void ClassAdapter::createLog(LogModel *lm)
					{
						//1：先读取文件的内容
						std::vector<LogModel*> list = this->readLogFile();
						//2：加入新的日志对象
						list.push_back(lm);
						//3：重新写入文件
						this->writeLogFile(list);
					}

					std::vector<LogModel*> ClassAdapter::getAllLog()
					{
						return this->readLogFile();
					}

					void ClassAdapter::removeLog(LogModel *lm)
					{
						//1：先读取文件的内容
						std::vector<LogModel*> list = this->readLogFile();
						//2：删除相应的日志对象
						list.remove(lm);
						//3：重新写入文件
						this->writeLogFile(list);
					}

					void ClassAdapter::updateLog(LogModel *lm)
					{
						//1：先读取文件的内容
						std::vector<LogModel*> list = this->readLogFile();
						//2：修改相应的日志对象
						for(int i=0;i<list.size();i++)
						{
							if(list.at(i)->getLogId()->equals(lm->getLogId()))
							{
								list.set(i, lm);
								break;
							}
						}
						//3：重新写入文件
						this->writeLogFile(list);
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/adapter/example5/ClassAdapter.java
package cn.javass.dp.adapter.example5;

import java.util.List;

/**
 * 类适配器对象
 */
public class ClassAdapter extends LogFileOperate implements LogDbOperateApi {

    public ClassAdapter(String logFilePathName) {
        super(logFilePathName);
    }

    public void createLog(LogModel lm) {
        //1：先读取文件的内容
        List<LogModel> list = this.readLogFile();
        //2：加入新的日志对象
        list.add(lm);
        //3：重新写入文件
        this.writeLogFile(list);
    }

    public List<LogModel> getAllLog() {
        return this.readLogFile();
    }

    public void removeLog(LogModel lm) {
        //1：先读取文件的内容
        List<LogModel> list = this.readLogFile();
        //2：删除相应的日志对象
        list.remove(lm);
        //3：重新写入文件
        this.writeLogFile(list);
    }

    public void updateLog(LogModel lm) {
        //1：先读取文件的内容
        List<LogModel> list = this.readLogFile();

        //2：修改相应的日志对象
        for (int i = 0; i < list.size(); i++) {
            if (list.get(i).getLogId().equals(lm.getLogId())) {
                list.set(i, lm);
                break;
            }
        }

        //3：重新写入文件
        this.writeLogFile(list);
    }
}
